### ISSUE TITLE
Migration of BNS1 namespaces v1 to v2 for BNS2

### ISSUE THREAD

**Issue by lgalabru** [blockstack/stacks-blockchain#2303]
Price functions can be updated in BNS 2, however we should assume that the keypairs that were used for creating namespaces under BNS 1 have been destroyed (PBC case), or lost.
As a result, we should be including a BTC -> STX conversion of the price function, in the migration process (or converting namespace v1 to namespace v2 if a helper is available). 
This can be done either python side (probably more elegant), or worst case, rust side (a bit patchy, since the version is not part of the dump).
Note: if integrated python side, it's fine if this patch is only available on the node creating the v1 dump.

![image](https://user-images.githubusercontent.com/87777/104083633-44124580-520e-11eb-9c9d-2be438cd460d.png)

**Comment by zone117x** [comment]
Unless @diwakergupta is okay with making more changes to the Stacks 1.0 image, we have to do this in in the Stacks 2.0 import. Also, we'd need special handling for at least the `btc` namespace as it already has price parameters set to be 2 STX / name in Stacks 2.0.

**Comment by diwakergupta** [comment]
> Unless @diwakergupta is okay with making more changes to the Stacks 1.0 image, we have to do this in in the Stacks 2.0 import. Also, we'd need special handling for at least the `btc` namespace as it already has price parameters set to be 2 STX / name in Stacks 2.0.

If doing it in Python is significantly simpler & easier, I'm OK making a patch and only updating the core-exporter node.

**Comment by zone117x** [comment]
@lgalabru these are the current namespaces and parameters exported
```
namespace_id,address,buckets,base,coeff,nonalpha_discount,no_vowel_discount,lifetime
app,SP24TC3Y58XKHZ7GX0N69X50BFYD9ECSR8PGAE3H6,6;5;4;3;2;1;0;0;0;0;0;0;0;0;0;0,4,250,4,4,0
blockstack,SP355AMV5R2A5C7VQCF4YPPAS05W93HTB68574W7S,7;6;5;4;3;2;1;1;1;1;1;1;1;1;1;1,4,250,4,4,0
btc,SP24TC3Y58XKHZ7GX0N69X50BFYD9ECSR8PGAE3H6,1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1,1000,200,1,1,262800
graphite,SP3X2W6GTGVVG9RW78F1MT7M9AQFS2HWV2JETXPG2,6;5;4;3;2;1;0;0;0;0;0;0;0;0;0;0,4,250,4,4,52595
helloworld,SP3TJ4WRTXW0YFWF1TWKWCG7E6Z6MZDTWAVB6SW5V,15;14;13;12;11;10;9;8;7;6;5;4;3;2;1;0,2,4,2,5,0
id,SP364R1Y0XYC8PYXEK0PE3V752RAJB2GNWF28WP90,6;5;4;3;2;1;0;0;0;0;0;0;0;0;0;0,4,250,4,4,52595
miner,SP24TC3Y58XKHZ7GX0N69X50BFYD9ECSR8PGAE3H6,0;0;0;0;0;0;0;0;0;0;0;0;0;0;0;0,1,0,1,1,0
mining,SP24TC3Y58XKHZ7GX0N69X50BFYD9ECSR8PGAE3H6,6;5;4;3;2;1;0;0;0;0;0;0;0;0;0;0,4,250,4,4,0
podcast,SPHYHD2ZJ11HWRDKVZ160QX7FD7PCVNT2N9CT2H6,1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1,40,250,1,1,0
stacking,SP24TC3Y58XKHZ7GX0N69X50BFYD9ECSR8PGAE3H6,6;5;4;3;2;1;0;0;0;0;0;0;0;0;0;0,4,250,4,4,0
stacks,SP24TC3Y58XKHZ7GX0N69X50BFYD9ECSR8PGAE3H6,6;5;4;3;2;1;0;0;0;0;0;0;0;0;0;0,4,250,4,4,0
stx,SP24TC3Y58XKHZ7GX0N69X50BFYD9ECSR8PGAE3H6,6;5;4;3;2;1;0;0;0;0;0;0;0;0;0;0,4,250,4,4,0
```
I also have access to that version byte in the python. Can you give me some tips on which parameters here need to be modified? For example, should all namespaces with a `1` version byte have the `coeff` modified? And modified to what?

**Comment by zone117x** [comment]
As point of reference, the name `matt.id` price is currently `160000` micro-stx in Stacks 2.0.

**Comment by kantai** [comment]
For version `1` (and version `2` that are more than a year old, which both `helloworld` and `podcast` are) namespaces, the current pricing scheme in Stacks 1.0 multiplies the satoshi price by (650/15), so I think the coefficient should be modified to `floor((650/15) * coeff)`

For example, for `id`:

```
id,SP364R1Y0XYC8PYXEK0PE3V752RAJB2GNWF28WP90,6;5;4;3;2;1;0;0;0;0;0;0;0;0;0;0,4,250,4,4,52595
```

Would become:

```
id,SP364R1Y0XYC8PYXEK0PE3V752RAJB2GNWF28WP90,6;5;4;3;2;1;0;0;0;0;0;0;0;0;0;0,4,10833,4,4,52595
```

**Comment by diwakergupta** [comment]
Fixed in #2307
